packages/shared/src/components/notifications/NotificationItemAvatar.tsx

```diff
@@ -16,10 +17,22 @@ function NotificationItemAvatar({
 }: NotificationAvatar): ReactElement {
   if (type === NotificationAvatarType.Source) {
     return (
-      <SourceButton
-        link={targetUrl}
-        source={{ id: referenceId, handle: referenceId, name, image }}
-      />
+      <LinkWithTooltip
```

Comment:
The type will always be missing and is a required property for getting permalink. Making it not suitable for the `SourceButton` anymore so I had to build an element of its own for `Notifications`.

Reply:
I added the targetURL there that solves the notification issues :)

Reply:
Yes, but the type will produce an error now as I have updated the `Source` to have a `type` property. So the build fails.

Reply:
I'd really go back to what it was, and optionally create the URL if passed link is empty.
It takes away overhead

Reply:
Me as well, the typescript is just not accepting it because the `SourceButton` property for `source` has a `type`. And from the notification, we won't know what `Source` type it is.

Reply:
That's why I said we can make the type default to machine for backwards compatability.

Reply:
Backwards compatibility is fine, the issue is the future notifications, the source type will always be `machine` if we do that since we don't have idea if it was from a notification source type of squad.
```
export interface Source {
  __typename?: string;
  id?: string;
  name: string;
  image: string;
  handle: string;
  type: SourceType;
  permalink: string;
}
```

Our notification avatar
```
export interface NotificationAvatar {
  type: NotificationAvatarType;
  image: string;
  name: string;
  targetUrl: string;
  referenceId: string;
}
```

Reply:
Yes, the notification never needs to know all the source information.
It always has a absolute URL which takes priority.

I'm not sure I'm following the issue.

Reply:
Yes, we are using the targetUrl. The problem becomes typescript. It is not accepting the `source` parameter because we don't have a `type` property. It is the build itself that rejects.

Reply:
Alright so we need to fix that, rather then just removing it.
Make it optional when it comes from notifications maybe?

Reply:
There is one option to keep the same component, it is if we are always sure permalink is going to have a value from the API. In that case, we will instead make it receive a custom object than `Source` object itself. WDYT?

Reply:
Not sure if it is possible to change props based on a single implementation. We will have to explicitly change its prop.